### ISSUE TITLE
You are being redirected / Javascript is required. Please enable javascript before you are allowed to see this page

### ISSUE THREAD

**Issue by AGDELTA** [Anorov/cloudflare-scrape#270]
Hello, I have the next code in shell script

#!/bin/bash

func_scrape() { V_SCRAPE_URL="$1"
cat <<EOF
# -*- coding: utf-8 -*-
import cfscrape
headers = {"user-agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/44.0.2403.89 Safari/537.36"}
scraper = cfscrape.create_scraper(delay=20)  # returns a CloudflareScraper instance
print scraper.get("$V_SCRAPE_URL", headers=headers).content
EOF
}

But where I executed, view the next error:

func_scrape "https://www.divxtotal3.net/series-16" | python
<html><title>You are being redirected...</title>
<noscript>Javascript is required. Please enable javascript before you are allowed to see this page.</noscript>
<script>var s={},u,c,U,r,i,l=0,a,e=eval,w=String.fromCharCode,sucuri_cloudproxy_js='',S='ZT0iOXN1Ii5zbGljZSgwLDEpICsgIjAiICsgU3RyaW5nLmZyb21DaGFyQ29kZSgweDM4KSArICJkIiArICdPNycuc2xpY2UoMSwyKSsgJycgKydPYTY3Jy5zdWJzdHIoMywgMSkgKyI1c3UiLnNsaWNlKDAsMSkgKyAiMnN1Ii5zbGljZSgwLDEpICsgImNzZWMiLnN1YnN0cigwLDEpICsgIjEiICsgICcnICsnJysiNW8iLmNoYXJBdCgwKSArICdkJyArICAiIiArIjEiLnNsaWNlKDAsMSkgKyAiNG4iLmNoYXJBdCgwKSArICIzIi5zbGljZSgwLDEpICsgIjRzdWN1ciIuY2hhckF0KDApKyI1aCIuY2hhckF0KDApICsgIjFzdWN1ciIuY2hhckF0KDApKyc2JyArICAgJycgKycnKyIwIiArICAnJyArJycrIjRzZWMiLnN1YnN0cigwLDEpICsgU3RyaW5nLmZyb21DaGFyQ29kZSgxMDApICsgImYiLnNsaWNlKDAsMSkgKyAiOXEiLmNoYXJBdCgwKSArICI5c3VjdXIiLmNoYXJBdCgwKSsgJycgKydHckxiJy5zdWJzdHIoMywgMSkgKyIwIiArICAnJyArIApTdHJpbmcuZnJvbUNoYXJDb2RlKDB4MzIpICsgIjRzdSIuc2xpY2UoMCwxKSArICJkc2VjIi5zdWJzdHIoMCwxKSArICIxc3UiLnNsaWNlKDAsMSkgKyAgJycgKyAKIjQiICsgIiIgKycnO2RvY3VtZW50LmNvb2tpZT0ncycrJ3VzdScuY2hhckF0KDApICsnYycrJ3UnKydzdWN1cicuY2hhckF0KDQpKyAnaScrJ18nKycnKydjc3VjdXJpJy5jaGFyQXQoMCkgKyAnbCcuY2hhckF0KDApKydvJysnJysndXN1Y3VyJy5jaGFyQXQoMCkrICdkJysnc3VjdXJpcCcuY2hhckF0KDYpKydyJysnJysnbycrJ3gnKyd5c3VjdXInLmNoYXJBdCgwKSsgJ19zdWMnLmNoYXJBdCgwKSsgJ3VzdScuY2hhckF0KDApICsnc3UnLmNoYXJBdCgxKSsnaXN1YycuY2hhckF0KDApKyAnZCcrJ3N1Y3VfJy5jaGFyQXQoNCkrICdzdWN1cmkzJy5jaGFyQXQoNikrJ2QnKycnKycxc3VjdScuY2hhckF0KDApICArJzhzdWN1cicuY2hhckF0KDApKyAnYycrJ3N1YycuY2hhckF0KDIpKydkJysnJysnMScrJ3NlJy5jaGFyQXQoMSkrIj0iICsgZSArICc7cGF0aD0vO21heC1hZ2U9ODY0MDAnOyBsb2NhdGlvbi5yZWxvYWQoKTs=';L=S.length;U=0;r='';var A='ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/';for(u=0;u<64;u++){s[A.charAt(u)]=u;}for(i=0;i<L;i++){c=s[S.charAt(i)];U=(U<<6)+c;l+=6;while(l>=8){((a=(U>>>(l-=8))&0xff)||(i<(L-2)))&&(r+=w(a));}}e(r);</script></html>

It is the equal error, if I execute with command curl

# curl -sk https://www.divxtotal3.net/series-16
<html><title>You are being redirected...</title>
<noscript>Javascript is required. Please enable javascript before you are allowed to see this page.</noscript>
<script>var s={},u,c,U,r,i,l=0,a,e=eval,w=String.fromCharCode,sucuri_cloudproxy_js='',S='eD0nMG1TYScuc3Vic3RyKDMsIDEpICsgJycgKycnKyI4Ii5zbGljZSgwLDEpICsgImJzdWN1ciIuY2hhckF0KDApKyI4bCIuY2hhckF0KDApICsgICcnICsgCiI5c3VjdXIiLmNoYXJBdCgwKSsnNycgKyAgICcnICsnMjEnLnNsaWNlKDEsMikrJzYnICsgICIyIiArICIiICsiZnN1Ii5zbGljZSgwLDEpICsgImNzdWN1ciIuY2hhckF0KDApKyIiICsndUdjJy5jaGFyQXQoMikrJz0zJy5zbGljZSgxLDIpK1N0cmluZy5mcm9tQ2hhckNvZGUoMHgzNCkgKyAnOScgKyAgJzcnICsgICAnJyArJzAnICsgICJjayIuY2hhckF0KDApICsgICcnICsgCiIwdyIuY2hhckF0KDApICsgImUiICsgImIiICsgIiIgKydhJyArICBTdHJpbmcuZnJvbUNoYXJDb2RlKDB4MzUpICsgIjQiLnNsaWNlKDAsMSkgKyAgJycgKyAKIjFzdWN1ciIuY2hhckF0KDApKyAnJyArJzMnICsgICI4bSIuY2hhckF0KDApICsgICcnICsgCiJkc3UiLnNsaWNlKDAsMSkgKyAiIiArU3RyaW5nLmZyb21DaGFyQ29kZSgweDY0KSArICdWeD5lJy5zdWJzdHIoMywgMSkgKyAnJyArIjFzdWN1ciIuY2hhckF0KDApKyJjIiArICAnJyArJyc7ZG9jdW1lbnQuY29va2llPSdzc3VjdXJpJy5jaGFyQXQoMCkgKyAndXMnLmNoYXJBdCgwKSsnYycrJ3VzdScuY2hhckF0KDApICsnc3VjdXJyJy5jaGFyQXQoNSkgKyAnaScrJ19zdScuY2hhckF0KDApICsnc3VjdXJpYycuY2hhckF0KDYpKydsJy5jaGFyQXQoMCkrJ29zdWN1Jy5jaGFyQXQoMCkgICsnc3UnLmNoYXJBdCgxKSsnc3VjdXJkJy5jaGFyQXQoNSkgKyAncCcuY2hhckF0KDApKydyJysnJysnc3VjdXJpbycuY2hhckF0KDYpKyd4c3VjdScuY2hhckF0KDApICArJ3lzJy5jaGFyQXQoMCkrJ18nKyd1JysndScrJ2knKydkJysnX3N1Y3VyJy5jaGFyQXQoMCkrICdlJysnc3VjdXI4Jy5jaGFyQXQoNSkgKyAnNicrJ2VzJy5jaGFyQXQoMCkrJzVzdWN1cmknLmNoYXJBdCgwKSArICczJy5jaGFyQXQoMCkrJzlzdScuY2hhckF0KDApICsnYnN1YycuY2hhckF0KDApKyAnZXMnLmNoYXJBdCgwKSsiPSIgKyB4ICsgJztwYXRoPS87bWF4LWFnZT04NjQwMCc7IGxvY2F0aW9uLnJlbG9hZCgpOw==';L=S.length;U=0;r='';var A='ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/';for(u=0;u<64;u++){s[A.charAt(u)]=u;}for(i=0;i<L;i++){c=s[S.charAt(i)];U=(U<<6)+c;l+=6;while(l>=8){((a=(U>>>(l-=8))&0xff)||(i<(L-2)))&&(r+=w(a));}}e(r);</script></html>

I am use the last version to cloudflare-scrape with python2.7

# pip show cfscrape
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7.
Name: cfscrape
Version: 2.0.7
Summary: A simple Python module to bypass Cloudflare's anti-bot page. See https://github.com/Anorov/cloudflare-scrape for more information.
Home-page: https://github.com/Anorov/cloudflare-scrape
Author: Anorov
Author-email: anorov.vorona@gmail.com
License: UNKNOWN
Location: /usr/local/lib/python2.7/dist-packages
Requires: requests
Required-by: 

The nodejs is the last version

# nodejs -v
v11.15.0

and the curl is

# curl --version
curl 7.52.1 (arm-unknown-linux-gnueabihf) libcurl/7.52.1 OpenSSL/1.0.2r zlib/1.2.8 libidn2/0.16 libpsl/0.17.0 (+libidn2/0.16) libssh2/1.7.0 nghttp2/1.18.1 librtmp/2.3
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtmp rtsp scp sftp smb smbs smtp smtps telnet tftp 
Features: AsynchDNS IDN IPv6 Largefile GSS-API Kerberos SPNEGO NTLM NTLM_WB SSL libz TLS-SRP HTTP2 UnixSockets HTTPS-proxy PSL 

My OS is raspbian strech.

Any ideas on how to solve the problem? Thanks

**Comment by pro-src** [comment]
Oh wow, I thought Cloudflare had retired that code a long time ago... Looks like it'll have to be added back in...

**Comment by pro-src** [comment]
I might have spoke too soon. I'm not sure if this is related to Cloudflare...
I'll try to poke around a bit more at a latter time.

**Comment by pro-src** [comment]
This isn't Cloudflare related.

There is some code to handle this in another Cloudflare scraping library of which I am a contributor. I was thinking about removing the bits that handles this, a method called `setCookieAndReload`. It wouldn't handle this particular case anyway because, as it turns out, this isn't Cloudflare related at all. 

I'm curious as to have often people encounter these but it's unlikely that the capabilities to handle this would be added here. Still, feel free to open another issue for a feature request.
